src/unitxt/task.py

```diff
@@ -99,9 +105,46 @@ def check_metrics_type(self) -> None:
                 f"metric's prediction type ({metric_prediction_type}) are different."
             )
 
+    def verify_defaults(self):
+        if self.defaults:
+            if not isinstance(self.defaults, dict):
+                raise ValueError(
+                    f"If specified, the 'defaults' must be a dictionary, "
+                    f"however, '{self.defaults}' was provided instead, "
+                    f"which is of type '{type(self.defaults)}'."
+                )
+
+            for default_name, default_value in self.defaults.items():
+                assert isinstance(default_name, str), (
+                    f"If specified, all keys of the 'defaults' must be strings, "
+                    f"however, the key '{default_name}' is of type '{type(default_name)}'."
+                )
+
+                val_type = self.inputs.get(default_name) or self.outputs.get(
+                    default_name
+                )
+
+                assert val_type, (
+                    f"If specified, all keys of the 'defaults' must refer to a chosen "
+                    f"key in either 'inputs' or 'outputs'. However, the name '{default_name}' "
+                    f"was provided which does not match any of the keys."
+                )
+
+                assert isoftype(default_value, parse_type_string(val_type)), (
+                    f"The value of '{default_name}' from the 'defaults' must be of "
+                    f"type '{val_type}', however, it is of type '{type(default_value)}'."
+                )
+
+    def set_default_values(self, instance: Dict[str, Any]) -> Dict[str, Any]:
+        if self.defaults:
+            instance.update(self.defaults)
```

Comment:
What happens if the value is already set in the instance? In this case, we don't want to value to be updated.

Reply:
Alright, I thought it should be overwritten anyway. So I'll change that